node_modules/@stdlib/iter/cartesian-power/benchmark/benchmark.js

```diff
@@ -0,0 +1,49 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2019 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+const bench = require( '@stdlib/bench' );
+const pkg = require( './../package.json' ).name;
+const iterCartesianPower = require( './../lib' );
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var arr = [ 'a', 'b', 'c' ];
+	var n = 2;
+	var iterator = iterCartesianPower( arr, n );
+	var v;
+
+	// Run the benchmark...
+	b.tic();
+	for ( var i = 0; i < b.iterations; i++ ) {
+		v = iterator.next();
+		if ( v.done ) {
+			b.fail( 'should iterate over all elements' );
+		}
+	}
+	b.toc();
+	if ( v.done ) {
+		b.pass( 'benchmark finished' );
+	} else {
+		b.fail( 'should have finished iteration' );
```

Comment:
This logic is not correct. You actually need to ensure that `v.done` does **not** return `true`. The point is to test per iteration performance, not how long it can generate all combinations. If folks want to generate all combos, they should just use `@stdlib/array/cartesian-power`. Instead, we want to know how long it takes a single iteration.